docs/controllers/infra-cluster.md

```diff
@@ -0,0 +1,24 @@
+# Infra cluster controller
+
+## Overview
+
+Infra cluster controller is responsible for managing InfrastructureCluster(AWSCluster, etc.) CRs. 
+The cluster object will represent the cloud infrastructure where current cluster where operator 
+is running because we treat this cluster as both management and workload.
```

Comment:
I suggest to rephrase this last sentence. I'm not sure that i've got it right...